### DISCUSSION TITLE
can we assign a weight matrix to polarizability during training?

### DISCUSSION THREAD

**Discussion by kechangming** [deepmodeling/deepmd-kit#3472] [General]
How can we modify loss/tensor.py to assign different weights to polar components during training? in our system, there is a two-order magnitude difference between the polarizability diagonal elements and off-diagonal elements, and the off-diagonal elements hold greater significance for our system. Hence, we need to increase the weight of the off-diagonal part during training.
![image](https://github.com/deepmodeling/deepmd-kit/assets/44281341/61ba4927-82e6-4135-a697-f6bbed0d1a2b)

**Comment by wanghan-iapcm** [comment]
Your are recommended to switch on the shift_diag option, which compute the average value (diag_shift) of the diagonal part from training data, fit the model with the trace free polar, then add the model output with the diag_shift.

**Reply by kechangming** [reply]
Thank you very much for your reply. However, our matrix is in the form of [[2, 0.01, 0.01], [0.01, 2, 0.01], [0.01, 0.01, 0.2]], and even after using the trace free polar, the training performance of the off-diagonal elements is still not satisfactory. May I ask how I can set the dimensions of the weight tensor if I want to use a weight tensor? I am currently unsure how to check the dimensions of t_atomic_polarizability.
![error](https://github.com/deepmodeling/deepmd-kit/assets/44281341/6f4a14e5-9377-42bb-b5f5-55b88b7a6418)
